backend/migration_tools/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,367 @@
+
+            # Clear the destination table if specified
+            if clear_table and dest_table:
+                logger.info(f"Clearing table '{dest_table}'")
+                try:
+                    start_time = time.time()
+                    dest_cursor.execute(f"DELETE FROM {dest_table}")
+                    dest_conn.commit()
+                    elapsed_time = time.time() - start_time
+                    logger.info(
+                        f"Table '{dest_table}' cleared successfully. Time taken: "
+                        f"{elapsed_time:.2f} seconds."
+                    )
+                except Exception as e:
+                    logger.error(f"Error clearing table '{dest_table}': {e}")
+                    dest_conn.rollback()
+                    raise
+
+            src_cursor.execute(src_query)
+            column_names = [desc[0] for desc in src_cursor.description]
+
+            self.migrate_rows(
+                migration_name,
+                src_cursor,
+                dest_cursor,
+                dest_conn,
+                dest_query,
+                column_names,
+                new_key_transaction,
+            )
+
+            # Adjust the auto-increment value
+            if dest_table and "id" in column_names:
+                self.adjust_auto_increment(dest_cursor, dest_table, dest_conn)
+
+            # Record the migration
+            self.record_migration(migration_name)
+            dest_conn.commit()
+            logger.info(f"Migration '{migration_name}' completed successfully")
+
+        src_cursor.close()
+        dest_cursor.close()
+        src_conn.close()
+        dest_conn.close()
+
+
+class Command(BaseCommand):
+    help = "Migrates data from v1 models to v2 models for all apps"
+
+    def handle(self, *args, **options):
+        v2_schema = "unstract_v2"
+        src_db_config = {
+            "dbname": settings.DB_NAME,
+            "user": settings.DB_USER,
+            "password": settings.DB_PASSWORD,
+            "host": settings.DB_HOST,
+            "port": settings.DB_PORT,
+        }
+        dest_db_config = {
+            "dbname": settings.DB_NAME,
+            "user": settings.DB_USER,
+            "password": settings.DB_PASSWORD,
+            "host": settings.DB_HOST,
+            "port": settings.DB_PORT,
+        }
+        schemas_to_migrate = os.getenv("SCHEMAS_TO_MIGRATE", None)
+        migration_query = UnstractMigration(v2_schema=v2_schema)
+
+        # Public tables
+        public_schema_migrations = migration_query.get_public_schema_migrations()
+        migrator = DataMigrator(
+            src_db_config, dest_db_config, v2_schema, batch_size=1000
+        )
+        migrator.migrate(public_schema_migrations)
+
+        if not schemas_to_migrate:
+            confirmation = (
+                input(
+                    "No specific schemas provided. This will migrate all schemas. "
+                    "Do you want to continue? (y/n): "
+                )
```

Comment:
@muhammad-ali-e when we are running these migrations with containers I don't think we can make an input. 

We could do something like below.

1. If `SCHEMAS_TO_MIGRATE` is empty we can simply skip the migration or raise an error with message.
2. If the `SCHEMAS_TO_MIGRATE` contains only one element and it is `_ALL_` , let's migrate everything.
@hari-kuriakose what do you suggest wrt this?

Reply:
Updated